### PR TITLE
upgraded to Riverpod 2 and Appwrite 1.1.0.

### PR DESCRIPTION
**Pull request by mandolf0**
Upgraded to Appwrite backend v 1.1.0 and Flutter_riverpod 2.x

This is my first PR.

Nice repo you got going on here @tsvillain .

## Appwrite v1 Note about Account
Account from appwrite.dart if used to fetch account, create user, update user, etc.
Account from models.dart is the "user model" which has userName, $id, email, and other fields.

Mando

### CODE REVIEW

**Comment by tsvillain** [comment]
@mandolf0 Thanks for updating the project. I will review the PR soon. I can see you are committing bug fixes, let me know when you are confident with the code so I can start the review process.
Thanks.

**Comment by mandolf0** [comment]
> @mandolf0 Thanks for updating the project. I will review the PR soon. I can see you are committing bug fixes, let me know when you are confident with the code so I can start the review process. Thanks.

It's ready.

File: frontend/lib/core/state/auth_state.dart [hunk 1]
```diff
@@ -6,32 +6,34 @@ import 'package:note_keeper/data/models/models.dart';
...
       : super(const AuthState.unauthenticated(isLoading: true)) {
     refresh();
+    state = state.copyWith(isLoading: false);
   }
 
   Future<void> refresh() async {
+    // from master repo
     try {
-      final user = await _authRepository.get();
+      final user = await ref.read(BackendDependency.account).get();
```
Comment (tsvillain) [inline frontend/lib/core/state/auth_state.dart]: @mandolf0 Any specific reason to directly call `.get()` from BackendDependency.account provider?

File: frontend/lib/data/repositories/database_repository_impl.dart [hunk 2]
```diff
@@ -11,27 +11,36 @@ import 'package:note_keeper/domain/repositories/repositories.dart';
...
+
+  //below is the working constructor. but needs to have user Id as field
+  DatabaseRepositoryImpl(this._database, this._authState, this.ref);
+  String get userId => _authState.user!.$id;
 
   static Provider<DatabaseRepositoryImpl> get provider =>
       _databaseRepositoryProvider;
-
-  final Database _database;
+  final Ref ref;
```
Comment (tsvillain) [inline frontend/lib/data/repositories/database_repository_impl.dart]: @mandolf0  ref is never used. no need to create Ref

File: frontend/lib/presentation/pages/authentication/login/login_page.dart [hunk 3]
```diff
@@ -1,11 +1,21 @@
...
 import 'package:flutter/material.dart';
 import 'package:flutter_riverpod/flutter_riverpod.dart';
 import 'package:note_keeper/core/navigation/routes.dart';
+import 'package:note_keeper/core/provider.dart';
+import 'package:note_keeper/core/state/state.dart';
 import 'package:note_keeper/presentation/pages/authentication/login/login_view_model.dart';
 import 'package:note_keeper/presentation/pages/authentication/widgets/text_field.dart';
 import 'package:routemaster/routemaster.dart';
 
+final _authProvider =
```
Comment (tsvillain) [inline frontend/lib/presentation/pages/authentication/login/login_page.dart]: @mandolf0 _authProvider and _authServiceProvider is unused. We should not have any unused/ required references.

File: frontend/lib/presentation/pages/authentication/login/login_view_model.dart [hunk 4]
```diff
@@ -8,6 +9,10 @@ import 'package:note_keeper/presentation/base_view_model.dart';
 
 final _loginViewModel = ChangeNotifierProvider((ref) => LoginViewModel(
     ref.read(Repository.auth), ref.read(AppState.auth.notifier)));
+final _authRepositoryProvider =
```
Comment (tsvillain) [inline frontend/lib/presentation/pages/authentication/login/login_view_model.dart]: @mandolf0 again not in use.

**Review by tsvillain — CHANGES_REQUESTED** [review]
Changes LGTM 🎉 and a Big Thanks 🙇 for updating the project with the latest packages.
Though a few unused codes don't break the app, we should remove them as we don't need them.
I know, I reviewed it after a big break you may be contributing to another project. If that's the case if you can not available to resolve them within 48 hours I will push the fix and will merge the PR.

Again Thanks for the PR, really appreciated.

File: frontend/lib/core/state/auth_state.dart (hunk 1 above)
Comment (mandolf0) [reply frontend/lib/core/state/auth_state.dart]: My mistake. Change reverted.

File: frontend/lib/data/repositories/database_repository_impl.dart (hunk 2 above)
Comment (mandolf0) [reply frontend/lib/data/repositories/database_repository_impl.dart]: It is needed in lines 12-14 to  later define the provider => in line 24;

File: frontend/lib/presentation/pages/authentication/login/login_page.dart (hunk 3 above)
Comment (mandolf0) [reply frontend/lib/presentation/pages/authentication/login/login_page.dart]: Revised to remove the unused reference.

File: frontend/lib/presentation/pages/authentication/login/login_view_model.dart (hunk 4 above)
Comment (mandolf0) [reply frontend/lib/presentation/pages/authentication/login/login_view_model.dart]: Revised to remove the unused reference.

File: frontend/lib/note_keeper.dart [hunk 5]
```diff
@@ -26,15 +26,20 @@ class _NoteKeeperState extends ConsumerState<NoteKeeper> {
...
+      /// hide keyboard when tapping outside of the input fields.
+      onTap: () => FocusManager.instance.primaryFocus?.unfocus(),
+      child: MaterialApp.router(
+        debugShowCheckedModeBanner: false,
+        scaffoldMessengerKey: globalScaffold,
+        themeMode: ThemeMode.dark,
+        darkTheme: ThemeData.dark(),
+        routeInformationParser: const RoutemasterParser(),
+        routerDelegate: RoutemasterDelegate(routesBuilder: (context) {
+          final isAuthenticated = ref.watch(_isAuthenticateProvider);
```
Comment (mandolf0) [inline frontend/lib/note_keeper.dart:39]: When we update the username, for example, the entire app reloads. It takes us from the profile page, shows the login view for a moment, then takes us back to the home page.

I was thinking we can somehow show a spinner loading and not really show that the entire app reloaded.

Comment (tsvillain) [reply frontend/lib/note_keeper.dart:39]: For this, you can create a new issue and let's merge this PR.

**Review by tsvillain — APPROVED** [review]

**Comment by tsvillain** [comment]
🎉 Congratulations @mandolf0 on your first PR merge.

**Comment by mandolf0** [comment]
> 🎉 Congratulations @mandolf0 on your first PR merge.

Thank you.

**Pull request merged**